compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/reduce-reactive-deps/hoist-deps-diff-ssa-instance1.expect.md

```diff
@@ -0,0 +1,96 @@
+
+## Input
+
+```javascript
+import {identity, shallowCopy, Stringify, useIdentity} from 'shared-runtime';
+
+type HasA = {kind: 'hasA'; a: {value: number}};
+type HasC = {kind: 'hasC'; c: {value: number}};
+function Foo({cond}: {cond: boolean}) {
+  let x: HasA | HasC = shallowCopy({kind: 'hasA', a: {value: 2}});
+  /**
+   * This read of x.a.value is outside of x's identifier mutable
+   * range + scope range. We mark this ssa instance (x_@0) as having
+   * a non-null object property `x.a`.
+   */
+  Math.max(x.a.value, 2);
+  if (cond) {
+    x = shallowCopy({kind: 'hasC', c: {value: 3}});
+  }
+
+  /**
+   * Since this x (x_@2 = phi(x_@0, x_@1)) is a different ssa instance,
+   * we cannot safely hoist a read of `x.a.value`
+   */
+  return <Stringify val={!cond && [(x as HasA).a.value + 2]} />;
+}
+export const FIXTURE_ENTRYPOINT = {
+  fn: Foo,
+  params: [{cond: false}],
+  sequentialRenders: [{cond: false}, {cond: true}],
+};
+
+```
+
+## Code
+
+```javascript
+import { c as _c } from "react/compiler-runtime";
+import { identity, shallowCopy, Stringify, useIdentity } from "shared-runtime";
+
+type HasA = { kind: "hasA"; a: { value: number } };
+type HasC = { kind: "hasC"; c: { value: number } };
+function Foo(t0) {
+  const $ = _c(7);
+  const { cond } = t0;
+  let t1;
+  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+    t1 = shallowCopy({ kind: "hasA", a: { value: 2 } });
+    $[0] = t1;
+  } else {
+    t1 = $[0];
+  }
+  let x = t1;
+
+  Math.max(x.a.value, 2);
+  if (cond) {
+    let t2;
+    if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
+      t2 = shallowCopy({ kind: "hasC", c: { value: 3 } });
+      $[1] = t2;
+    } else {
+      t2 = $[1];
+    }
+    x = t2;
+  }
+  let t2;
+  if ($[2] !== cond || $[3] !== x) {
```

Comment:
This is finally correct -- fixed by #30573 ! Note that before removing `LeaveSSA`, we incorrectly take a dependency on `x` here as we weren't able to distinguish reassignment instances (see [previous repro on previous PR](https://react-compiler-playground-fboeug9zp-fbopensource.vercel.app/#N4Igzg9grgTgxgUxALhAHQHYDMobgFwEsIMACAMQggAphS4SATUgX2VLoY0fYCMqANggCGZFgEoOmUqSH5SAD3YAJYWACCpAD6lVYAMKkAvKTAALYQIEQA7vogAHAJ61SAa0Ld2AcgsbvADSkwux0AG6WUAjsAEysrOIA3NKkAPQAVOkppOmkACpmhGCkMCLMEFiKAHTCVRECUaRFpND4YISMCC2VCt7FHQgYRFiECDCkALZQ+MK8Qtm5MKIA5l0A1KYMDl1LGKtVpADqXRPCMG6k+IXFYGDCTRhgM3hd1AoA+gACAAySaqQWMKeZYLYKkDAkAC0GCgVhavAAVggCKQHDBHGN8E5SAADBQ1HFVBapFIAWWEVyqpwUbxqdUiCCCMSSKUIlWoXEYkmA2UUxlMFistnszlcHi8pF8an0gXooVI9Si7AAzPEJMkyKxMCkMllNTlSABlTyIS7XPlvL5xEwOQqWn5BD6fACM4kkzXujDZWDGg3kt3unieokQAVBNi6cFEEP9wh9AmxZggRXk91KwnKlTxdMVCBxxJSpXwsDIAB5DfgYMC2dj6kZgABCTmkABkLdIAG18bVc6QNjEALosNIAPg1LBALCAA))